src/setup_node_env/node_version_validator.js

```diff
@@ -28,27 +28,201 @@
  * under the License.
  */
 
+/* Note:
+ *   This file uses ES5 in order to provide meaningful output even if an old Node.js runtime is used.
+ */
+
 var pkg = require('../../package.json');
 
-// Note: This is written in ES5 so we can run this before anything else
-// and gives support for older NodeJS versions
-var currentVersion = (process && process.version) || null;
-var rawRequiredVersion = (pkg && pkg.engines && pkg.engines.node) || null;
-var requiredVersion = rawRequiredVersion ? 'v' + rawRequiredVersion : rawRequiredVersion;
-var currentVersionMajorMinorPatch = currentVersion.match(/^v(\d+)\.(\d+)\.(\d+)/);
-var requiredVersionMajorMinorPatch = requiredVersion.match(/^v(\d+)\.(\d+)\.(\d+)/);
-var isVersionValid =
-  currentVersionMajorMinorPatch[1] === requiredVersionMajorMinorPatch[1] &&
-  currentVersionMajorMinorPatch[2] === requiredVersionMajorMinorPatch[2] &&
-  parseInt(currentVersionMajorMinorPatch[3], 10) >= parseInt(requiredVersionMajorMinorPatch[3], 10);
-
-// Validates current the NodeJS version compatibility when OpenSearch Dashboards starts.
-if (!isVersionValid) {
-  var errorMessage =
-    `OpenSearch Dashboards was built with ${requiredVersion} and does not support the current Node.js version ${currentVersion}. ` +
-    `Please use Node.js ${requiredVersion} or a higher patch version.`;
-
-  // Actions to apply when validation fails: error report + exit.
-  console.error(errorMessage);
-  process.exit(1);
+var pkgEngineNodeVersion = pkg && pkg.engines && pkg.engines.node;
+if (!pkgEngineNodeVersion || typeof pkgEngineNodeVersion !== 'string') {
+  console.error(
+    'OpenSearch Dashboards did not report its required version of the Node.js runtime. ' +
+      'Please revert any changes that might have been made to the package.json file and try again.'
+  );
+  process.exit(22);
+}
+
+/* Basic semver parsing: This is a very limited subset of what semver supports where only a single comparator, composed
+ * of an operator and a version, is supported.
+ * [https://github.com/npm/node-semver/blob/cb1ca1d5480a6c07c12ac31ba5f2071ed530c4ed/README.md#ranges]
+ *
+ * The supported operators are:
+ *   >    Greater than
+ *   >=   Greater than or equal to
+ *   =    Equal
+ *   ~    Tilde ranges: Allows patch changes if a minor version is specified but if only a major version is specified,
+ *          it allows minor changes.
+ *   ^    Caret ranges: Allows patch and minor updates when major is non-zero (and we will never have that).
+ *
+ * Note: If no operator is specified, equality is assumed.
+ */
+var requiredParts = pkgEngineNodeVersion.match(
+  /^\s*(>=?|\^|~|=)?\s*(\d+)(?:\.(\d+))?(?:\.(\d+))?(?:[^\d.].*)?$/
+);
+if (requiredParts === null) {
+  console.error(
+    'OpenSearch Dashboards did not report its required version of the Node.js runtime in a valid ' +
+      'format. Please revert any changes that might have been made to the package.json file and try again.'
```

Comment:
Can we just extract this string out to a variable instead of hard-coding twice?

Reply:
Done.